### DISCUSSION TITLE
Is PlusToolkit  support the pose tracking of a robot arm?

### DISCUSSION THREAD

**Discussion by Barneyhuuu** [PlusToolkit/PlusLib#916] [Q&A]
Is PlusToolkit  support the pose tracking of a robot arm, like universal collaboration robot (UR5e)? We plan to use the robot arm pose tracking instead of a electromagnetic tracking device, but I don't see any robot on the support list.

**Comment by ungi** [comment]
For now, UR robots are recommended to be used using OpenIGTLink. PLUS supports any device that can send their tracking data through OpenIGTLink. An OpenIGTLinkBridge node is being developed for ROS2 to relay the information both ways between PLUS and UR or Slicer and UR. But it's not finished yet. Are you interested in helping in the development?

**Reply by Barneyhuuu** [reply]
Sorry to reply you so late. Our experiment was on hold for a while due to the COVID-19. Anyway, with the help of your advice, we recently tested ROS-IGtL-Bridge from the tutorials on OpenIGTLink  and it worked! But rather than we expected, the different kinds of information were passed from ROS to 3D Slicer OpenIGtLink module without PLUS involved. So, is this right? Or there is another way to send pose tracking information from ROS to 3D slicer with Plustoolkit? Because we are now using PLUS to import ultrasound image and electromagnetic tracking device to 3D Slicer. Thank you for your help!
As for the development, yes! I'd love to be involved, but the fact that I'm a spine surgeon and I have little background in programming and software development, I'm not sure I can help. If there's anything I can do, I'd be happy to！

**Reply by Barneyhuuu** [reply]
![Uploading 2022-03-31 下午8.58.37.png…]()

<img width="1352" alt="2022-03-31 下午8 55 34" src="https://user-images.githubusercontent.com/93598282/161061824-fa4c1a16-a7dd-4730-8753-a560577c8f14.png">

**Reply by Barneyhuuu** [reply]
<img width="1432" alt="2022-03-31 下午8 58 37" src="https://user-images.githubusercontent.com/93598282/161062025-b17f4fbb-6a3f-4932-977e-6f348c9541f7.png">
